src/java/picard/analysis/GcBiasUtils.java

```diff
@@ -0,0 +1,44 @@
+package picard.analysis;
+
+/**
+ * Created by kbergin on 9/23/15.
+ */
+public class GcBiasUtils {
+
+    /////////////////////////////////////////////////////////////////////////////
+    // Calculates GC as a number from 0 to 100 in the specified window.
+    // If the window includes more than five no-calls then -1 is returned.
+    /////////////////////////////////////////////////////////////////////////////
+    public static int calculateGc(final byte[] bases, final int startIndex, final int endIndex, final CalculateGcState state) {
+        if (state.init) {
+            state.init = false;
+            state.gcCount = 0;
+            state.nCount = 0;
+            for (int i = startIndex; i < endIndex; ++i) {
+                final byte base = bases[i];
+                if (base == 'G' || base == 'C') ++state.gcCount;
+                else if (base == 'N') ++state.nCount;
+            }
+        } else {
+            final byte newBase = bases[endIndex - 1];
+            if (newBase == 'G' || newBase == 'C') ++state.gcCount;
+            else if (newBase == 'N') ++state.nCount;
+
+            if (state.priorBase == 'G' || state.priorBase == 'C') --state.gcCount;
+            else if (state.priorBase == 'N') --state.nCount;
```

Comment:
This is sort of tangential, but could this method be modified to use ```SequenceUtil.basesEqual``` instead of direct ```==``` ? It would be cleaner, and you wouldn't have to call ```StringUtil.toUpperCase(refBases)``` elsewhere (which might speed things up, though perhaps not enough to be noticeable).